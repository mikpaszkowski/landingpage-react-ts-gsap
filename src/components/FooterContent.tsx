import * as React from "react";

export const FooterContent = () => {};
