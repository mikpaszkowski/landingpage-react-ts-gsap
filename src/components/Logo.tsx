import * as React from "react";
import { ReactComponent as Ellipse2 } from "../assets/images/svg-images/ellipse-2.svg";
import { ReactComponent as Ellipse1 } from "../assets/images/svg-images/ellipse-1.svg";
import styled from "styled-components";
import { colors } from "../styles/styleConstants";

const LogoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
`;

const LogoText = styled.p`
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: ${(props: LogoProps) =>
    props.inverse ? "white" : colors.primaryFont};
`;

const Ellipse1Absolute = styled(Ellipse1)`
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
`;

interface LogoProps {
  inverse?: boolean | null;
}

export const Logo = ({ inverse }: LogoProps) => {
  return (
    <LogoWrapper>
      <Ellipse1Absolute />
      <Ellipse2 />
      <LogoText inverse={inverse}>Medli</LogoText>
    </LogoWrapper>
  );
};
