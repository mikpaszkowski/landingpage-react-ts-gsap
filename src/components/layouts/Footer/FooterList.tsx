import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { lists } from "../../staticContents/footer/footerLists";
import { device } from "../../../styles/responsive";

const List = styled.ul`
  & > li {
    color: ${({ theme }) => theme.palette.primary.fontLight};
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    &:first-child {
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface FooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: FooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </List>
  );
};
