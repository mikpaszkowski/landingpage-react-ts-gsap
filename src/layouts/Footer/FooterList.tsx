import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { lists } from "../../data/staticContents/footer/footerLists";
import { device } from "../../styles/responsive";

const List = styled.ul`
  margin-bottom: 3.1rem;
  & > li {
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    & a {
      color: ${({ theme }) => theme.palette.primary.fontLight};
      text-decoration: none;
      font-size: 1.8rem;
      line-height: 2.7rem;
    }

    &:first-child {
      text-transform: uppercase;
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface IFooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: IFooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map(({ url, label }, index) => (
        <li key={index}>
          <a href={url}>{label}</a>
        </li>
      ))}
    </List>
  );
};
