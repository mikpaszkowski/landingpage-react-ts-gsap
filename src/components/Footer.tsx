import * as React from "react";
import { Logo } from "./Logo";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${colors.footerBackground};
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${colors.primaryLight};
  opacity: 0.5;
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
