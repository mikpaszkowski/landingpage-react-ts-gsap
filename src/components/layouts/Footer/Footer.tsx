import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { FooterContent } from "./FooterContent";
import { device } from "../../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.mobileMedium} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
