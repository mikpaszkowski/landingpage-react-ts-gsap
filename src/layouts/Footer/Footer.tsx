import * as React from "react";
import { Logo } from "../../components/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { FooterContent } from "./FooterContent";
import { device } from "../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.tablet} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  display: block;
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileLarge} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  display: block;
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileLarge} {
    display: none;
  }

  @media ${device.tablet} {
    right: 2.5rem;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
