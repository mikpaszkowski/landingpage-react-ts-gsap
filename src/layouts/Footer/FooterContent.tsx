import * as React from "react";
import styled from "styled-components";
import { FooterList } from "./FooterList";
import { EmailInputContainer } from "../../components/EmailInputContainer";
import { device } from "../../styles/responsive";

const FooterContentWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin: 5.1rem 0 8rem 0;

  @media ${device.tabletLarge} {
    flex-direction: column-reverse;
    justify-content: flex-start;
    margin: 5.6rem 0 0 0;
  }
`;

const FooterListWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  ul:nth-child(2) {
    margin-right: 14rem;
  }

  ul:first-child {
    margin-right: 5.8rem;
  }

  @media ${device.tabletLarge} {
    flex-direction: column;
    justify-content: flex-start;
  }
`;

const footerLists = [
  {
    id: 1,
    listName: "products",
  },
  {
    id: 2,
    listName: "company",
  },
  {
    id: 3,
    listName: "help",
  },
];

export const FooterContent = () => {
  return (
    <FooterContentWrapper>
      <FooterListWrapper>
        {footerLists.map(({ id, listName }) => (
          <FooterList key={id} listName={listName} />
        ))}
      </FooterListWrapper>
      <EmailInputContainer />
    </FooterContentWrapper>
  );
};
