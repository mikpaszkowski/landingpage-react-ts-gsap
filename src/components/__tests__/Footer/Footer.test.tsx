import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { Footer } from "../../layouts/Footer/Footer";
import { FooterContent } from "../../layouts/Footer/FooterContent";
import { Logo } from "../../ReusableComponents/Logo/Logo";

afterEach(() => {
  cleanup();
});

test("check render correctly <Footer /> component", () => {
  const tree = renderWithLightTheme(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check Footer component", () => {
  const wrapper = mountWithDarkTheme(<Footer />);

  test("should contain FooterContent component", () => {
    expect(wrapper.find(FooterContent).length).toEqual(1);
  });

  describe("should contain INVERSED Logo component", () => {
    test("single Logo component", () => {
      expect(wrapper.find(Logo).length).toEqual(1);
    });
    test("should have prop inverse set to true", () => {
      expect(wrapper.find(Logo).prop("inverse")).toEqual(true);
    });
  });

  test("should contain p tag element", () => {
    expect(wrapper.find(Footer).find("p").at(2).text()).toEqual(
      "Therms & Conditions"
    );
  });
});
