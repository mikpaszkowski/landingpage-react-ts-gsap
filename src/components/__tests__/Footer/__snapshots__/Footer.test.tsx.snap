// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <Footer /> component 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
  cursor: pointer;
}

.c3 {
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: white;
}

.c2 {
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
}

.c6 {
  margin-bottom: 3.1rem;
}

.c6 > li {
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  list-style-type: none;
}

.c6 > li a {
  color: #E5E5E5;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  line-height: 2.7rem;
}

.c6 > li:first-child {
  text-transform: uppercase;
  color: #7C8176;
  margin-bottom: 1.9rem;
}

.c8 {
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: 'Reem Kufi',sans-serif;
  color: #7C8176;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid #7C8176;
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;
}

.c10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 1.8rem;
  font-style: normal;
  font-weight: normal;
  line-height: 2.7rem;
  color: #131E07;
  border: none;
  border-radius: 4px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c10::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 4px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c10:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c10:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c7 {
  position: relative;
}

.c7 > p {
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  margin-bottom: 1.6rem;
  color: #7C8176;
}

.c11 {
  position: absolute;
  right: 0.8rem;
  bottom: 0.8rem;
  padding: 0.3rem 1.6rem 0.4rem 1.6rem;
  border-radius: 4px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 5.1rem 0 8rem 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c5 ul:nth-child(2) {
  margin-right: 14rem;
}

.c5 ul:first-child {
  margin-right: 5.8rem;
}

.c0 {
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: #131E07;
}

.c12 {
  display: block;
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;
}

.c13 {
  display: block;
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: 'Reem Kufi',sans-serif;
  color: #E5E5E5;
  opacity: 0.5;
}

@media (max-width:375px) {
  .c6 {
    margin-bottom: 3.1rem;
  }
}

@media (max-width:375px) {
  .c8 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c7 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c11 {
    position: relative;
    width: 100%;
    right: 0;
    bottom: 0;
    font-size: 2rem;
    padding: 1.6rem 8rem 1.6rem 8rem;
    margin: 1.6rem 0 6.3rem 0;
  }
}

@media (max-width:1110px) {
  .c9 {
    width: 100%;
  }
}

@media (max-width:1110px) {
  .c4 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 5.6rem 0 0 0;
  }
}

@media (max-width:1110px) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:768px) {
  .c0 {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
}

@media (max-width:525px) {
  .c12 {
    display: none;
  }
}

@media (max-width:525px) {
  .c13 {
    display: none;
  }
}

@media (max-width:768px) {
  .c13 {
    right: 2.5rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
    >
      ellipse-1.svg
    </svg>
    <svg>
      ellipse-2.svg
    </svg>
    <p
      className="c3"
    >
      Medli
    </p>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <ul
        className="c6"
      >
        <li>
          products
        </li>
        <li>
          <a
            href="#"
          >
            Updates
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Security
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Help
          </a>
        </li>
      </ul>
      <ul
        className="c6"
      >
        <li>
          company
        </li>
        <li>
          <a
            href="#"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Join Us
          </a>
        </li>
      </ul>
      <ul
        className="c6"
      >
        <li>
          help
        </li>
        <li>
          <a
            href="#"
          >
            Talk to Support
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Support Docs
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            API Docs
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            System Status
          </a>
        </li>
      </ul>
    </div>
    <div
      className="c7"
    >
      <p>
        Try Medli for free
      </p>
      <input
        className="c8 c9"
        placeholder="Enter your email"
      />
      <button
        className="c10 c11"
        fontSize="1.8rem"
      >
        Get started
      </button>
    </div>
  </div>
  <div
    className="c12"
  />
  <p
    className="c13"
  >
    Therms & Conditions
  </p>
</div>
`;
